diff --git a/src/gallium/auxiliary/gallivm/lp_bld_misc.cpp b/src/gallium/auxiliary/gallivm/lp_bld_misc.cpp
index ce8ca02..f741d69 100644
--- a/src/gallium/auxiliary/gallivm/lp_bld_misc.cpp
+++ b/src/gallium/auxiliary/gallivm/lp_bld_misc.cpp
@@ -331,9 +331,7 @@ lp_build_fill_mattrs(std::vector<std::string> &MAttrs)
       llvm::sys::getHostCPUFeatures(features);
    #endif
 
-   for (llvm::StringMapIterator<bool> f = features.begin();
-        f != features.end();
-        ++f) {
+   for (auto f = features.begin(); f != features.end(); ++f) {
       MAttrs.push_back(((*f).second ? "+" : "-") + (*f).first().str());
    }
 #elif DETECT_ARCH_X86 || DETECT_ARCH_X86_64
